tests/backup/backup_partial_test.go

```diff
@@ -1226,3 +1215,230 @@ var _ = Describe("{PartialBackupWithLowerStorkVersion}", func() {
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, namespace := range backedUpNamespaces {
+				namespaceMapping[namespace] = namespace + RandomString(4)
+			}
+			for _, backupName := range backupNames {
+				restoreName := fmt.Sprintf("%s-%s-%s", "default", backupName, RandomString(4))
+				log.InfoD("Restoring from the [%s] backup with namespaceMapping [%v]", backupName, namespaceMapping)
+				err = CreatePartialRestoreWithValidation(ctx, restoreName, backupName, namespaceMapping, make(map[string]string), DestinationClusterName, BackupOrgID, scheduledAppContexts, failedVolumes)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Creation and Validation of restore [%s]", restoreName))
+				restoreNames = append(restoreNames, restoreName)
+			}
+		})
+
+		Step("Restoring the backed up namespaces in same namespace on source with retain", func() {
+			log.InfoD("Restoring the backed up namespaces in same namespace on source with retain")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, backupName := range backupNames {
+				restoreName := fmt.Sprintf("%s-%s-%s", "retain", backupName, RandomString(4))
+				log.InfoD("Restoring from the [%s] backup ", backupName)
+				err := CreatePartialRestoreWithReplacePolicyWithValidation(restoreName, backupName, namespaceMapping, DestinationClusterName, BackupOrgID, ctx, make(map[string]string), 1, scheduledAppContexts, failedVolumes)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Creation and Validation of restore [%s]", restoreName))
+				restoreNames = append(restoreNames, restoreName)
+			}
+		})
+
+		Step("Restoring the backed up namespaces in same namespace on source with replace", func() {
+			log.InfoD("Restoring the backed up namespaces in same namespace on source with replace")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, backupName := range backupNames {
+				restoreName := fmt.Sprintf("%s-%s-%s", "replace", backupName, RandomString(4))
+				err := CreatePartialRestoreWithReplacePolicyWithValidation(restoreName, backupName, namespaceMapping, DestinationClusterName, BackupOrgID, ctx, make(map[string]string), 2, scheduledAppContexts, failedVolumes)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Creating restore [%s]", restoreName))
+				restoreNames = append(restoreNames, restoreName)
+			}
+		})
+	})
+
+	JustAfterEach(func() {
+		defer EndPxBackupTorpedoTest(scheduledAppContexts)
+
+		defer func() {
+			log.InfoD("switching to default context")
+			err := SetClusterContext("")
+			log.FailOnError(err, "failed to SetClusterContext to default cluster")
+			AzureEnvironmentEnvVar[AzureEnvironmentEnvVarName] = ""
+			err = UpdateDriverWithEnvVariable(AzureEnvironmentEnvVar)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Resetting AZURE_ENVIRONMENT to default value"))
+			log.Infof("Unsetting BACKUP_TYPE env variable")
```

Comment:
are we not supposed to reset the BACKUP_TYPE rather than unset

Reply:
We are unsetting it from the value we set  so that for next use, it will fetch from env itself.